sunpy/net/dataretriever/sources/goes.py

```diff
@@ -58,18 +58,26 @@ def _get_url_for_timerange(self, timerange, **kwargs):
             Data type to return for the particular GOES satellite. Supported
             types depend on the satellite number specified. (default = xrs_2s)
         """
-
+        start_date = timerange.start
+        end_date = timerange.end
```

Comment:
like the comment before... are these variables needed? they are just used to calculate the number of days... which `timerange` provides them already, no?